mobile/src/main/java/org/hawkular/client/android/activity/LoginActivity.java

```diff
@@ -181,44 +163,32 @@ private void showError(@StringRes int errorMessage) {
     }
 
 
-    private static final class AuthorizeCallback extends AbstractActivityCallback<String> {
+    private static final class AuthorizeCallback implements Callback<List<Metric>> {
 
-        @Override
-        public void onSuccess(String authorization) {
-            LoginActivity activity = (LoginActivity) getActivity();
+        LoginActivity activity;
 
-            activity.setUpBackendCommunication(new Persona("hawkular"));
-            ((LoginActivity) getActivity()).succeed();
+        public AuthorizeCallback(LoginActivity activity) {
+            this.activity = activity;
         }
 
-        @Override
-        public void onFailure(Exception e) {
-            Timber.d(e, "Authorization failed.");
+        @Override public void onResponse(Call<List<Metric>> call, Response<List<Metric>> response) {
+            if (response.isSuccessful()){
+                getActivity().succeed();
+            }
```

Comment:
@pg301 That loop is because I have not handled else condition in here. Please correct it.